src/shared/istiopolicy/creator.go

```diff
@@ -98,6 +102,80 @@ func (c *Creator) Create(
 	return nil
 }
 
+func (c *Creator) UpdateIntentsStatus(
+	ctx context.Context,
+	clientIntents *v1alpha2.ClientIntents,
+	clientServiceAccount string,
+) error {
+	err := c.labelWithServiceAccount(ctx, clientIntents, clientServiceAccount)
+	if err != nil {
+		return err
+	}
+
+	return c.updateStatusSharedServiceAccount(ctx, clientIntents.Namespace, clientServiceAccount)
+}
+
+func (c *Creator) labelWithServiceAccount(ctx context.Context, clientIntents *v1alpha2.ClientIntents, clientServiceAccount string) error {
+	serviceAccountLabelValue, ok := clientIntents.Labels[v1alpha2.OtterizeClientServiceAccountLabel]
+	if ok && serviceAccountLabelValue == clientServiceAccount {
+		return nil
+	}
+
+	labeledIntents := clientIntents.DeepCopy()
+	if labeledIntents.Labels == nil {
+		labeledIntents.Labels = make(map[string]string)
+	}
+
+	labeledIntents.Labels[v1alpha2.OtterizeClientServiceAccountLabel] = clientServiceAccount
+	err := c.client.Patch(ctx, labeledIntents, client.MergeFrom(clientIntents))
+	if err != nil {
+		logrus.WithError(err).Errorln("Failed labeling intent with service account name")
+		c.recorder.RecordWarningEventf(clientIntents, ReasonIntentsLabelFailed, "Failed labeling intent with service account name: %s", err.Error())
+		return err
+	}
+
+	logrus.Infof("Labeling intent %s with service account name %s", clientIntents.Name, clientServiceAccount)
+
+	c.recorder.RecordNormalEventf(clientIntents, ReasonServiceAccountFound, "Client intents service account found %s", clientServiceAccount)
+	return nil
+}
+
+func (c *Creator) updateStatusSharedServiceAccount(ctx context.Context, namespace string, clientServiceAccountName string) error {
+	var ClientIntentsList v1alpha2.ClientIntentsList
+	err := c.client.List(ctx, &ClientIntentsList, client.InNamespace(namespace), client.MatchingLabels{
+		v1alpha2.OtterizeClientServiceAccountLabel: clientServiceAccountName,
+	})
+
+	if err != nil {
+		return err
+	}
+
+	logrus.Infof("Found %d intents with service account %s", len(ClientIntentsList.Items), clientServiceAccountName)
+	HasSharedServiceAccount := len(ClientIntentsList.Items) > 1
+	for _, intents := range ClientIntentsList.Items {
+		if intents.Status != nil && intents.Status.HasSharedServiceAccount == HasSharedServiceAccount {
+			continue
+		}
+
+		updatedIntents := intents.DeepCopy()
+		if updatedIntents.Status == nil {
+			updatedIntents.Status = &v1alpha2.IntentsStatus{}
+		}
+
+		updatedIntents.Status.HasSharedServiceAccount = HasSharedServiceAccount
+		err = c.client.Status().Patch(ctx, updatedIntents, client.MergeFrom(&intents))
+		if err != nil {
+			return err
+		}
+
+		if HasSharedServiceAccount {
+			c.recorder.RecordWarningEventf(updatedIntents, ReasonSharedServiceAccount, "Service account %s is shared by multiple clients", clientServiceAccountName)
```

Comment:
What is the outcome? The message should say that this means that other services will also be able to gain access because of this intent, and it should also say which ones (based on the list of client intents)